src/Webservices/Factory.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Factory for Joomla Webservices.
+ *
+ * @package    Webservices
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Webservices\Webservices;
+
+use Joomla\DI\Container;
+use Joomla\Registry\Registry;
+
+/**
+ * Factory for webservices.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Factory
+{
+	/**
+	 * Gets instance of helper object class if it exists.
+	 *
+	 * @param   string  $version  Webservice version.
+	 * @param   string  $client   Webservice client ('administrator' or 'site').
+	 * @param   string  $name     Webservice name.
+	 * @param   string  $path     Webservice path.
+	 * 
+	 * @return  mixed It will return Api helper class or false if it does not exist.
+	 *
+	 * @since   1.2
+	 */
+	public static function getHelper($version, $client, $name, $path)
+	{
+		static $apiHelper = null;
+
+		// If we already have it, return it.
+		if (!is_null($apiHelper))
+		{
+			return $apiHelper;
+		}
+
+		$helperFile = ConfigurationHelper::getWebserviceHelper($client, strtolower($name), $version, $path);
+
+		if (file_exists($helperFile))
+		{
+			require_once $helperFile;
+		}
+
+		$webserviceName = preg_replace('/[^A-Z0-9_\.]/i', '', $name);
+		$helperClassName = 'JApiHalHelper' . ucfirst($client) . ucfirst(strtolower($webserviceName));
+
+		if (!class_exists($helperClassName))
+		{
+			return false;
+		}
+
+		$apiHelper = new $helperClassName;
+
+		return $apiHelper;
+	}
+
+	/**
+	 * Get a profile object.
+	 * 
+	 * @param   Container  $container     Dependency injection container.
+	 * @param   string     $clientName    Client name (eg. 'administrator' or 'site').
+	 * @param   string     $resourceName  Name of resource for which the profile is sought.
+	 * @param   string     $version       Version of the resource profile sought.
+	 * @param   string     $operation     Operation name (eg. 'read', 'update').
+	 * 
+	 * @return  Profile
+	 */
+	public static function getProfile(Container $container, $clientName, $resourceName, $version, $operation)
+	{
+		$db = $container->get('db');
```

Comment:
You should inject the database object here imho as that's the only dependency that is in the container. We should be passing around the container as little as possible (in a perfect world it would only exist in the application)